### PR TITLE
Add therapist availabilty functions

### PR DESCRIPTION
**Pull request by aloomdar**

### CODE REVIEW

File: src/data-access/therapist.ts [hunk 1]
```diff
@@ -35,3 +35,57 @@ export async function listTherapistByPatient({
     .limit(limit)
     .offset(offset);
 }
+
+export async function therapistAvailability({
```
Comment (jesuskcapellan) [inline src/data-access/therapist.ts]: name should be something like listAvailableSlots or listAvailabillity

File: src/data-access/therapist.ts [hunk 2]
```diff
@@ -35,3 +35,57 @@ export async function listTherapistByPatient({
...
+    .where(
+      and(
+        eq(availability.therapistId, therapistId),
+        eq(availability.availableDate, date),
+        eq(availability.isBooked, false),
+      ),
+    );
+}
+
+export async function bookAvailabilitySlot({
```
Comment (jesuskcapellan) [inline src/data-access/therapist.ts]: name should be bookAvailableSlot

File: src/db/schema.ts [hunk 3]
```diff
@@ -265,3 +267,20 @@ export const messages = createTable("message", {
     .default(sql`CURRENT_TIMESTAMP`)
     .$onUpdate(() => sql`CURRENT_TIMESTAMP`),
 });
+
+export const availability = createTable("availability", {
+  availabilityId: serial("availability_id").primaryKey(),
+  therapistId: integer("therapist_id")
+    .notNull()
+    .references(() => users.userId),
+  availableDate: date("available_date").notNull(),
```
Comment (jesuskcapellan) [inline src/db/schema.ts]: nit: I feel "date" is appropriate here but it's up to you

**Review by jesuskcapellan — COMMENTED** [review]
Looks good minus a few things!

**Review by jesuskcapellan — APPROVED** [review]

**Review by DavidZamora35 — APPROVED** [review]

**Pull request merged**